pkg/apis/pravega/v1alpha1/pravegacluster_types.go

```diff
@@ -25,6 +25,9 @@ const (
 	// DefaultPravegaVersion is the default tag used for for the Pravega
 	// Docker image
 	DefaultPravegaVersion = "0.4.0"
+
+	// Default Domain Name
+	DefaultDomainName = "pravega.io"
```

Comment:
Don't use this as a default value.   Instead, conditionally apply the external DNS annotation based on whether a domain was provided.

Reply:
Sure.